elastic/apm-server
# model: Align `labels` field with ECS

## Motivation/summary

Updates all data streams to use the ecs defined `labels` for values that
are `string` or `bool`, and adds a new field for `numeric_labels` which
stores all numeric labels as `scaled_float`, using the `scaling_factor`
we were using for indexes.

There's a single exception for the `metrics-apm.internal` data stream,
due to the fact that `dynamic` is set to `strict`, which disables the
externally defined ecs labels dynamic ability. For that data stream, the
field is defined as a standard field.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Install the integration.
2. Ingest some data with both numeric, string and bool labels.
3. Verify that the string and bool labels are under the `labels` field
4. Verify that numeric (floats and ints) are stored under `numeric_labels`.

## Related issues

Closes #3873
